src/Serilog/Formatting/Display/LogEventPropertiesValue.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright 2017 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Serilog.Events;
+
+namespace Serilog.Formatting.Display
+{
+    class LogEventPropertiesValue : LogEventPropertyValue
+    {
+        readonly MessageTemplate _template;
+        readonly IReadOnlyDictionary<string, LogEventPropertyValue> _properties;
+
+        public LogEventPropertiesValue(MessageTemplate template, IReadOnlyDictionary<string, LogEventPropertyValue> properties)
+        {
+            _template = template;
+            _properties = properties;
+        }
+
+        public override void Render(TextWriter output, string format = null, IFormatProvider formatProvider = null)
+        {
+            output.Write('{');
+
+            var delim = "";
+            foreach (var kvp in _properties)
+            {
+                if (_template.NamedProperties.Any(x => x.PropertyName == kvp.Key))
```

Comment:
Since `NamedProperties` is an array, this can use indexer-based access and avoid allocating a LINQ enumerator. (Keeping an eye on allocations, as they build up.)